sdk/python/kfp_tekton/compiler/_data_passing_rewriter.py

```diff
@@ -524,21 +527,35 @@ def big_data_passing_tasks(prname: str, task: dict, pipelinerun_template: dict,
                 task_output.get('name')))
             workspaces_parameter = '$(workspaces.%s.path)/%s/%s/%s' % (
                 task_name, BIG_DATA_MIDPATH, "$(context.taskRun.name)", task_output.get('name'))
-            # For child nodes to know the taskrun name, it has to pass to results via /tekton/results emptydir
-            if not appended_taskrun_name:
-                copy_taskrun_name_step = _get_base_step('output-taskrun-name')
-                copy_taskrun_name_step['command'].append('echo -n "$(context.taskRun.name)" > "$(results.taskrun-name.path)"')
-                task['taskSpec']['results'].append({"name": "taskrun-name", "type": "string"})
-                task['taskSpec']['steps'].append(copy_taskrun_name_step)
-                _append_original_pr_name_env(task)
-                appended_taskrun_name = True
+            if env.get('OUTPUT_BIG_DATA_PATH', 'false').lower() == 'true':
+            # For child nodes to know the taskrun output path, it has to pass to results via /tekton/results emptydir
+                if not appended_taskrun_path_step:
+                    appended_taskrun_path_step = _get_base_step('output-taskrun-path')
+                if len(appended_taskrun_path_step['command']) <= 2:
+                    appended_taskrun_path_step['command'].append('')
+                appended_taskrun_path_step['command'][-1] += 'echo -n "%s/%s/%s" > $(results.%s-path.path)\n' % \
+                                                    (BIG_DATA_MIDPATH, "$(context.taskRun.name)", task_output.get('name'),
+                                                    task_output.get('name'))
+                task['taskSpec']['results'].append({"name": "%s-path" % (task_output.get('name')), "type": "string"})
+                task['taskSpec']['steps'].append(appended_taskrun_path_step)
```

Comment:
Looks like a new `output-taskrun-path` step is added for each subsequent result.

Reply:
thanks, I moved it to the post task output condition